### DISCUSSION TITLE
Multiple programs in the MINDSTORMS robot inventor hub

### DISCUSSION THREAD

**Discussion by JobadSayas** [pybricks/support#1003] [General]
When you install pybrick's firmware in MINDSTORMS robot inventor hub, would you still be able to have multiple programs installed directly in the hub? and is there a way to select what slot number to use? or how does it actually work?

**Comment by dlech** [comment]
The current Pybricks firmware (v3.2.x) only allows a single "program". However, this program can consist of multiple files and you can create your own menu in that program to simulate the "slots" of the official LEGO firmware. 

See https://github.com/pybricks/support/issues/861

**Comment by laurensvalk** [comment]
We've just added a tutorial that also answers this question. It works both with Python and block-based coding.

[![spike-hubmenu-yt](https://github.com/pybricks/support/assets/12326241/eeba4d24-4d68-40fd-8718-71e11e6634cf)](https://pybricks.com/project/spike-hub-menu/)
